app/src/main/java/com/nowak/wjw/simplecompass/data/location/LocationApiHandler.java

```diff
@@ -74,7 +67,20 @@ public void stopLocationUpdates() {
     public void requestLocationUpdates() throws SecurityException {
```

Comment:
to jest mega słabe ? Gdzie obsługa błędów ? IDE  nawet jest tak mądre że ci podpowiada niżej że wywołanie requestLocationUpdates jest niebezpieczne, dopóki ktoś nie zezwolił na użycie rzeczy które chcesz dotykać.
 if(ContextCompat.checkSelfPermission(context,ACCESS_COARSE_LOCATION)!=PERMISSION_GRANTED){
            return;
        }
i 2 zostawiam już tobie. niejako przekazujesz tam gdzie używasz tego api więc można go tu przekazać. a jeśli chcesz to wydzielić to gdzieś indziej sprawdzić i taką informację zapisać.

Reply:
a to mam wstrzyknąć context do tej klasy, czy dodać do metody parametr boolean hasRequiredPermissions i przekazać tego booleana z fragmentu?